### PR TITLE
Master wowl concurrency aab

### PR DESCRIPTION
**Pull request by aab-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/pivot/pivot_model.js [hunk 1]
```diff
@@ -380,11 +384,17 @@ export class PivotModel extends Model {
             context: {},
             domain: [],
             groupBy: [],
+        };
 
+        this.derivedSearchParams = {
```
Comment (aab-odoo) [inline addons/web/static/src/views/pivot/pivot_model.js:389]: in meta?

File: addons/web/static/src/views/pivot/pivot_model.js [hunk 2]
```diff
@@ -611,8 +611,9 @@ export class PivotModel extends Model {
     /**
      * Swap the pivot columns and the rows. It is a synchronous operation.
      */
-    flip() {
-        this._cancelPreviousOperation();
+    async flip() {
```
Comment (aab-odoo) [inline addons/web/static/src/views/pivot/pivot_model.js:625]: weird comment

File: addons/web/static/src/views/helpers/model.js [hunk 3]
```diff
@@ -48,7 +65,7 @@ function getSearchParams(props) {
     for (const key of SEARCH_KEYS) {
         params[key] = props[key];
     }
-    return params;
+    return JSON.parse(JSON.stringify(params));
```
Comment (aab-odoo) [inline addons/web/static/src/views/helpers/model.js:68]: do this in searchModel

File: addons/web/static/src/views/helpers/model.js [hunk 4]
```diff
@@ -19,6 +22,7 @@ export class Model extends EventBus {
     constructor(env, params, services) {
         super();
         this.env = env;
```
Comment (aab-odoo) [inline addons/web/static/src/views/helpers/model.js:24]: this.useSampleModel = false

File: addons/web/static/src/search/with_search/with_search.js [hunk 5]
```diff
@@ -83,6 +83,11 @@ export class WithSearch extends Component {
         for (const key of OTHER_SEARCH_KEYS) {
             componentProps.info[key] = this.searchModel[key];
         }
+        if (this.props.globalState) {
+            const globalState = { ...this.props.globalState };
+            delete globalState.searchModel;
+            componentProps.globalState = globalState;
+        }
```
Comment (aab-odoo) [inline addons/web/static/src/search/with_search/with_search.js:90]: move to View

File: addons/web/static/src/search/with_search/with_search.js [hunk 6]
```diff
@@ -83,6 +83,11 @@ export class WithSearch extends Component {
         for (const key of OTHER_SEARCH_KEYS) {
             componentProps.info[key] = this.searchModel[key];
         }
+        if (this.props.globalState) {
+            const globalState = { ...this.props.globalState };
+            delete globalState.searchModel;
```
Comment (aab-odoo) [inline addons/web/static/src/search/with_search/with_search.js:88]: not necessary

File: addons/web/static/src/views/helpers/model.js [hunk 7]
```diff
@@ -68,18 +85,51 @@ export function useModel(ModelClass, params, options = {}) {
...
-
     onWillUpdateProps((nextProps) => {
-        const searchParams = getSearchParams(nextProps);
-        searchParams.useSampleModel = false;
-        return model.load(searchParams);
+        useSampleModel = false;
+        return load(nextProps);
     });
+
+    useSetupView({
```
Comment (aab-odoo) [inline addons/web/static/src/views/helpers/model.js:125]: useSetupAction

File: addons/web/static/src/search/search_model.js [hunk 8]
```diff
@@ -427,7 +427,7 @@ export class SearchModel extends EventBus {
         if (!this._context) {
             this._context = this._getContext();
         }
-        return this._context;
+        return JSON.parse(JSON.stringify(this._context));
```
Comment (kebeclibre) [inline addons/web/static/src/search/search_model.js:430]: there is a deepCopy function in utils/object

Comment (aab-odoo) [reply addons/web/static/src/search/search_model.js:430]: :+1:

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by ged-odoo** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**